### PR TITLE
Add autogenerated luigi client api docs

### PR DESCRIPTION
**Pull request by jesusreal**
**Description**

Changes proposed in this pull request:

- Replace current Luigi Client API docu with an inline, autogenerated documentation

**Related issue(s)**

### CODE REVIEW

File: client/luigi-client.js [hunk 1]
```diff
@@ -279,30 +289,31 @@
...
         /**
-         * Shows a spinner with backdrop to block the micro front-end frame
+         * Adds a backdrop with a loading indicator for the micro front-end frame. This overrides the `loadingIndicator.enabled` setting.
          */
         showLoadingIndicator: function showLoadingIndicator() {
           window.parent.postMessage({ msg: 'luigi.show-loading-indicator' }, '*');
         },
         /**
-         * Hides the spinner
+         * Removes the loading indicator. Use it after calling {@link #showLoadingIndicator showLoadingIndicator()} or to hide the indicator when you use the `loadingIndicator.hideAutomatically: false` Node configuration.
```
Comment (maxmarkus) [inline client/luigi-client.js]: Can loadingIndicator.hideAutomatically be linked to the configuration?

File: client/luigi-client.js [hunk 2]
```diff
@@ -279,30 +289,31 @@
...
     /**
-     * Manage UX specific options.
+     * Use the UX Manager to manage the appearance in Luigi.
      */
     uxManager: function uxManager() {
       return {
+        /** @lends uxManager */
         /**
-         * Shows a spinner with backdrop to block the micro front-end frame
+         * Adds a backdrop with a loading indicator for the micro front-end frame. This overrides the `loadingIndicator.enabled` setting.
```
Comment (maxmarkus) [inline client/luigi-client.js]: Can `loadingIndicator.enabled` be linked to the configuration?

File: client/luigi-client.js [hunk 3]
```diff
@@ -39,205 +37,220 @@
...
+          };
+          window.parent.postMessage(navigationOpenMsg, '*');
         },
 
         /**
-         * Sets the current navigation context, which is then be used by navigate function
-         * Usage: linkManager.fromContext("currentTeam").navigate("path")
-         * @param {object} navigationContext
-         * */
+         * Sets the current navigation context to that of a specific parent Node that has the `navigationContext` field declared in its navigation configuration. This navigation context is then used by navigate function.
```
Comment (maxmarkus) [inline client/luigi-client.js]: Can we link `navigationContext` to the configuration documentation?

File: client/luigi-client.js [hunk 4]
```diff
@@ -39,205 +37,220 @@
     });
   }
 
-  var Luigi = {};
-  var currentContext = {};
+  // actual Luigi code
+  var luigiInitialized = false
   var defaultContextKeys = ['context', 'internal', 'nodeParams', 'pathParams'];
-  var _contextUpdated;
+  var currentContext = defaultContextKeys.reduce(function(acc, key) {
```
Comment (maxmarkus) [inline client/luigi-client.js]: `Array.reduce` is not available with ES5

File: client/luigi-client.js (hunk 1 above)
Comment (jesusreal) [reply client/luigi-client.js]: Linked to `navigation-configuration.md#nodes`

File: client/luigi-client.js (hunk 2 above)
Comment (jesusreal) [reply client/luigi-client.js]: Linked to `navigation-configuration.md#nodes`

File: client/luigi-client.js (hunk 3 above)
Comment (jesusreal) [reply client/luigi-client.js]: Linked to `navigation-configuration.md#nodes`

File: client/luigi-client.js (hunk 4 above)
Comment (jesusreal) [reply client/luigi-client.js]: Actually it is :) https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce#Specifications

Comment (maxmarkus) [reply client/luigi-client.js]: OMG, never seen it before ES6. Thanks for the info!

File: client/README.md [hunk 5]
```diff
@@ -24,3 +24,10 @@ or, if you are not using any bundler, Luigi is also available as a global object
 window.LuigiClient
 ```
 You can see the Luigi Client in action by running the [Angular example application](/core/examples/luigi-sample-angular).
+
+
+## Generate documentation
+# Validate and generate documentation
+npm install
+npm run docu
+```
```
[the remark this answers is not in the document]
Comment (jesusreal) [reply client/README.md]: done

File: docs/luigi-client-api.md [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+There are various parameters and functions available to Luigi pertaining the lifecycle of listeners, navigation Nodes and Event data.
+
+### addInitListener
+
+Registers a listener that is called with a context object as soon as Luigi is instantiated. Defer your application bootstrap if you are dependent on authentication data from Luigi.
+
+#### Parameters
+
+-   `initFn` **[function](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Statements/function)** the function that will be called once Luigi is initialized
```
[the remark this answers is not in the document]
Comment (jesusreal) [reply docs/luigi-client-api.md]: done

File: docs/luigi-client-api.md [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+
+Returns the configuration object of the active navigation Node.
+
+Returns **[Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Object)** node parameters
+
+### getPathParams
+
+Returns the dynamic path parameters of the active URL.
+
+Returns **[Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Object)** path parameters
```
[the remark this answers is not in the document]
Comment (jesusreal) [reply docs/luigi-client-api.md]: for parameters and returns we don´t use dots. Should we?

File: docs/luigi-client-api.md [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+
+### navigate
+
+Navigates to the given path in the hosting Luigi app. Contains either a full absolute path or a relative path without a leading slash that uses the active route as a base. This is a classical navigation.
+
+#### Parameters
+
+-   `path` **[string](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String)** path to be navigated to
+-   `sessionId` **[string](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String)** current Luigi sessionId
+-   `preserveView` **[boolean](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Boolean)** Preserve a view by setting it to true. It keeps the current view opened in the background and opens the new route in a new frame. You must use the function [goBack()](#goBack) to navigate back afterwards. You can use this feature at unlimited levels.The preserved views are discarded as soon as the standard [navigate()](#navigate) function is in use in place of [goBack()](#goBack) .
```
[the remark this answers is not in the document]
Comment (jesusreal) [reply docs/luigi-client-api.md]: done

File: docs/luigi-client-api.md [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+
+-   `navigationContext` **[Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Object)** 
+
+#### Examples
+
+```javascript
+LuigiClient.linkManager().fromContext('project').navigate('/settings')
+```
+
+Returns **[linkManager](#linkmanager)** link manager instance
```
[the remark this answers is not in the document]
Comment (jesusreal) [reply docs/luigi-client-api.md]: done

File: docs/luigi-client-api.md [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+
+Returns the context object. Usually it is not required as the [addContextUpdateListener()](#addContextUpdateListener) receives the same values.
+
+Returns **{idToken: [string](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String), sessionId: [string](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String), currentEnvironmentId: [string](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String)}** node parameters
+
+### getNodeParams
+
+Returns the configuration object of the active navigation Node.
+
+Returns **[Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Object)** node parameters
```
[the remark this answers is not in the document]
Comment (jesusreal) [reply docs/luigi-client-api.md]: done

File: docs/luigi-client-api.md [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+LuigiClient.linkManager.fromContext("currentTeam").withParams({foo: "bar"}).navigate("path")
+```
+
+Returns **[linkManager](#linkmanager)** link manager instance.
+
+### hasBack
+
+Checks if there are one or more preserved views. Can be used to show a back button.
+
+Returns **[boolean](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Boolean)** a boolean with the information if there is a preserved view available to which a user can return..
```
Comment (bszwarc) [inline docs/luigi-client-api.md]: Two dots at the end :)

Comment (jesusreal) [reply docs/luigi-client-api.md]: done

**Pull request merged**